controller/tests/controller_server/utils_test.py

```diff
@@ -40,32 +40,37 @@ def test_validate_secrets(self):
         with self.assertRaises(ValidationException):
             utils.validate_secret(secrets)
 
-    def test_validate_volume_capabilities(self):
-        caps = Mock()
-        caps.mount = Mock()
-        caps.mount.fs_type = "ext4"
+    def test_validate_file_system_volume_capabilities(self):
+        cap = Mock()
+        cap.mount = Mock()
+        cap.mount.fs_type = "ext4"
         access_types = csi_pb2.VolumeCapability.AccessMode
-        caps.access_mode.mode = access_types.SINGLE_NODE_WRITER
+        cap.access_mode.mode = access_types.SINGLE_NODE_WRITER
+        cap.HasField.return_value = True
 
-        utils.validate_csi_volume_capabilties([caps])
+        utils.validate_csi_volume_capabilties([cap])
 
         with self.assertRaises(ValidationException):
             utils.validate_csi_volume_capabilties([])
 
-        caps.mount.fs_type = "ext41"
+        cap.mount.fs_type = "ext4dummy"
         with self.assertRaises(ValidationException):
-            utils.validate_csi_volume_capabilties([caps])
+            utils.validate_csi_volume_capabilties([cap])
 
-        caps.mount.fs_type = "ext4"
-        caps.access_mode.mode = access_types.SINGLE_NODE_READER_ONLY
+        cap.mount.fs_type = "ext4"
+        cap.access_mode.mode = access_types.SINGLE_NODE_READER_ONLY
         with self.assertRaises(ValidationException):
-            utils.validate_csi_volume_capabilties([caps])
+            utils.validate_csi_volume_capabilties([cap])
 
+    def test_validate_raw_block_volume_capabilities(self):
         caps = Mock()
-        caps.mount = None
-        caps.access_mode.mode = access_types.SINGLE_NODE_READER_ONLY
-        with self.assertRaises(ValidationException):
-            utils.validate_csi_volume_capabilties([caps])
+        caps.block = Mock()
+        access_types = csi_pb2.VolumeCapability.AccessMode
+        caps.access_mode.mode = access_types.SINGLE_NODE_WRITER
+        # First call of HasField returns True and second - False
```

Comment:
As Robert C. Martin once wrote: "Comments are always failures".
replace comment with indicative `return_values_in_LIFO_order = [False, True]`.
if it's still not clear enough for you, also wrap it in a function: `set_return_values(function_mock=caps.HasField, for_first_call=True, for_second_call=False)`